### PR TITLE
snekim: init at 1.2.0

### PR DESCRIPTION
**Pull request by auroraanna**
###### Description of changes

Added the snekim  package and one of it's dependencies, nimraylib-now.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by auroraanna** [comment]
This is my follow up pr to https://github.com/NixOS/nixpkgs/pull/194822#event-7655616423

**Comment by auroraanna** [comment]
The snekim executable gives this error tho:
```console
> ./result/bin/snekim
INFO: Initializing raylib 4.2
INFO: Supported raylib modules:
INFO:     > rcore:..... loaded (mandatory)
INFO:     > rlgl:...... loaded (mandatory)
INFO:     > rshapes:... loaded (optional)
INFO:     > rtextures:. loaded (optional)
INFO:     > rtext:..... loaded (optional)
INFO:     > rmodels:... loaded (optional)
INFO:     > raudio:.... loaded (optional)
WARNING: GLFW: Error: 65542 Description: GLX: Failed to load GLX
WARNING: GLFW: Failed to initialize Window
FATAL: Failed to initialize Graphic Device
```
when not run with ``steam-run`

**Comment by auroraanna** [comment]
It runs without issues with `steam-run ./result/bin/snekim`

**Comment by auroraanna** [comment]
> It with `steam-run ./result/bin/snekim`, tho the nim's std/appdirs thinks the home directory is at `/homeless-shelter`?

That was fixed by not determining the home directory path at compile time.

**Comment by auroraanna** [comment]
> The snekim executable gives this error tho:
> 
> ```
> > ./result/bin/snekim
> INFO: Initializing raylib 4.2
> INFO: Supported raylib modules:
> INFO:     > rcore:..... loaded (mandatory)
> INFO:     > rlgl:...... loaded (mandatory)
> INFO:     > rshapes:... loaded (optional)
> INFO:     > rtextures:. loaded (optional)
> INFO:     > rtext:..... loaded (optional)
> INFO:     > rmodels:... loaded (optional)
> INFO:     > raudio:.... loaded (optional)
> WARNING: GLFW: Error: 65542 Description: GLX: Failed to load GLX
> WARNING: GLFW: Failed to initialize Window
> FATAL: Failed to initialize Graphic Device
> ```
> 
> when not run with `steam-run`

This was fixed by using raylib from nixpkgs (dynamic lib instead of static).

**Comment by auroraanna** [comment]
Oops, didn't mean to commit structorizer there, fixed it.

File: pkgs/games/snekim/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "annaaurora";
+    repo = "snekim";
+    rev = "v${version}";
+    sha256 = "sha256-Qgvq4CkGvNppYFpITCCifOHtVQYRQJPEK3rTJXQkTvI=";
+  };
+
+  buildInputs = [
```
Comment (ehmry) [inline pkgs/games/snekim/default.nix:17]: Please make these `propagatedBuildInputs` of `nimraylib-now`, unless you have a reason not to.

**Review by ghost — CHANGES_REQUESTED** [review]

File: pkgs/games/snekim/default.nix (hunk 1 above)
Comment (ehmry) [reply pkgs/games/snekim/default.nix:17]: @adamlwgriffiths unless you think the xorg stuff should go into the `raylib` `propagatedBuildInputs`.

Comment (adamlwgriffiths) [reply pkgs/games/snekim/default.nix:17]: I'm not well versed enough in Nix to say for sure, I know enough to be dangerous.
But from my naive glance at the difference, it seems like a good suggestion.

Comment (auroraanna) [reply pkgs/games/snekim/default.nix:19]: I read the nix manual (https://nixos.org/manual/nixpkgs/unstable/#ftn.id-1.4.2.5.5.2, and some other positions) but I still don't understand the difference between the buildInputs. I think uh… buildInputs is only for build-time inputs and propagatedBuildInputs is for build-time and run-time. Is that right?

Comment (auroraanna) [reply pkgs/games/snekim/default.nix:19]: I set `stricDeps = true` in the latest force push and it still works so, it's all only needed at build-time?

Comment (auroraanna) [reply pkgs/games/snekim/default.nix:19]: @ehmry

Comment (auroraanna) [reply pkgs/games/snekim/default.nix]: Actually, this package doesn't need to specify xorg dependencies because raylib's package already does so.

**Pull request merged**